arbor/simulation.cpp

```diff
@@ -514,6 +514,9 @@ void simulation::reset() {
 }
 
 time_type simulation::run(time_type tfinal, time_type dt) {
+    if (dt <= 0.0) {
+        throw arbor_exception("Finite time-step must be supplied.");
```

Comment:
Maybe an exception a tad more type specific? There's no `user_input_error` in `arbexcept?pp`, maybe this is a good occasion to add it.

Reply:
`pyarb_error` in `python/error.hpp` might be the ticket.

Reply:
In the end everything is the user's fault, but I think it's not very polite to tell them.
I decided to rip off `std` and make `domain_error` a thing.

Reply:
Here is the thing: shouldn't we be using that error type for all such errors in our interface?
Instead of creating a new error type, to be used for the one case in this change, we could make a separate issue to systematically address such errors.

Reply:
Sure, let's consider this the obvious start. We have checks on eg context and domain decomposition, but I am not aware of more place where this could/should go.